public/components/overview/home.tsx

```diff
@@ -3,159 +3,272 @@
+import { HOME_CONTENT_AREAS, HOME_PAGE_ID } from '../../plugin_helpers/plugin_overview';
+import { cardConfigs, GettingStartedConfig } from './components/card_configs';
+import { uiSettingsService } from '../../../common/utils';
+import { AddDashboardCallout } from './components/add_dashboard_callout';
+import { DashboardControls } from './components/dashboard_controls';
+import { SelectDashboardFlyout } from './components/select_dashboard_flyout';
+import { setNavBreadCrumbs } from '../../../common/utils/set_nav_bread_crumbs';
+import './index.scss';
 
+// Plugin IDs
 const alertsPluginID = 'alerting';
 const anomalyPluginID = 'anomalyDetection';
 
-// Plugin URLs
-const gettingStartedURL = 'observability-gettingStarted';
-const discoverURL = 'data-explorer';
-const metricsURL = 'observability-metrics';
-const tracesURL = 'observability-traces-nav#/traces';
-const alertsURL = 'alerting';
-const anomalyDetectionURL = 'anomaly-detection-dashboards';
-
-const checkIfPluginsAreInstalled = async (
-  setAlertsPluginExists: React.Dispatch<React.SetStateAction<boolean>>,
-  setAnomalyPluginExists: React.Dispatch<React.SetStateAction<boolean>>
-) => {
-  try {
-    const response = await fetch('../api/status', {
-      headers: {
-        'Content-Type': 'application/json',
-        'osd-xsrf': 'true',
-        'accept-language': 'en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7,zh-TW;q=0.6',
-        pragma: 'no-cache',
-        'sec-fetch-dest': 'empty',
-        'sec-fetch-mode': 'cors',
-        'sec-fetch-site': 'same-origin',
-      },
-      method: 'GET',
-      referrerPolicy: 'strict-origin-when-cross-origin',
-      mode: 'cors',
-      credentials: 'include',
-    });
-    const data = await response.json();
-
-    let alertsExists = false;
-    let anomalyExists = false;
+const uiSettingsKey = 'observability:defaultDashboard';
 
-    for (const status of data.status.statuses) {
-      if (status.id.includes(alertsPluginID)) {
-        alertsExists = true;
-      }
-      if (status.id.includes(anomalyPluginID)) {
-        anomalyExists = true;
-      }
-    }
+interface HomeProps extends RouteComponentProps {
+  parentBreadcrumbs: ChromeBreadcrumb[];
+  contentManagement: ContentManagementPluginStart;
+}
 
-    setAlertsPluginExists(alertsExists);
-    setAnomalyPluginExists(anomalyExists);
-  } catch (error) {
-    console.error('Error checking plugin installation status:', error);
-  }
-};
+let showFlyout: { (): void; (): void; (): void };
+const wrapper = { dashboardSelected: false };
+let startDate: string;
+let setStartDate: (start: string) => void;
+let endDate: string;
+let setEndDate: (end: string) => void;
+let dashboardTitle: string;
+let setDashboardTitle: (title: string) => void;
+let dashboardId: string;
+let setDashboardId: (id: string) => void;
 
-const navigateToApp = (appId: string, path: string) => {
+export const navigateToApp = (appId: string, path: string) => {
```

Comment:
Please use a different named function. This will mess up the autocomplete. Also this should be moved to utils or some folder outside home. If this is used my multiple code pieces.

Reply:
Renamed to navigateToApplication.

Reply:
removed it altogether we don't need this, I reverted back to calling the coreRef function directly.